### ISSUE TITLE
Runtime Dialog unexpected signal during runtime execution

### ISSUE THREAD

**Issue by lyimmi** [wailsapp/wails#257]
**Description**

When I try to use runtime.Dialog.SelectFile() the program exits with a panic.
My problem is that, "this" code below worked in the morning, I changed a few things in other parts of the code but not the directory selection.

I may be using it incorrectly, but that's how I interpreted the documentation. "A common pattern for the Runtime is to simply save it as part of the struct and use it when needed:"

**To Reproduce**
Steps to reproduce the behaviour:
Build the code below and call it from the frontend.

I use Vuetify2 if it matters.

Frontend
`
window.backend.MyRuntime.SelectDirectory()
`

GO:
```
package main

import (
	"github.com/leaanthony/mewn"
	"github.com/wailsapp/wails"
)

func basic() string {
	return "Hello World!"
}

// MyRuntime wails runtime
type MyRuntime struct {
	runtime *wails.Runtime
}

// WailsInit initialize wails
func (s *MyRuntime) WailsInit(r *wails.Runtime) error {
	s.runtime = r
	return nil
}

// SelectDirectory open a file selector dialog
func (s *MyRuntime) SelectDirectory() string {
	file := s.runtime.Dialog.SelectDirectory()
	return file
}

func main() {

	js := mewn.String("./frontend/dist/app.js")
	css := mewn.String("./frontend/dist/app.css")

	app := wails.CreateApp(&wails.AppConfig{
		Width:  1024,
		Height: 768,
		Title:  "test",
		JS:     js,
		CSS:    css,
		Colour: "#131313",
	})

	my := &MyRuntime{}

	app.Bind(my)
	app.Bind(basic)
	app.Run()
}

```

**Expected behaviour**
Open a directory selector dialog and return the selected folder's path as a sting.

**System Details**

| Name   | Value |
| ----- | ----- |
| Wails Version | v0.18.3 |
| Go Version    | go1.12.10 |
| Platform      | linux |
| Arch          | amd64 |
| GO111MODULE   | on |
| GCC           | 8.3.0 |
| Npm           | v10.16.3 |
| Node          | 6.9.0 |

**Additional context**
```
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7fad4f01009d]

runtime stack:
runtime.throw(0x8aac64, 0x2a)
	/snap/go/current/src/runtime/panic.go:617 +0x72
runtime.sigpanic()
	/snap/go/current/src/runtime/signal_unix.go:374 +0x4a9

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0x7b46d0, 0xc0003ff9f0, 0x0)
	/snap/go/current/src/runtime/cgocall.go:128 +0x5b fp=0xc0003ff9c0 sp=0xc0003ff988 pc=0x40675b
github.com/wailsapp/wails/lib/renderer/webview._Cfunc_CgoDialog(0x1845cf0, 0x100000000, 0x199c5e0, 0x1b9a060, 0x1c5e000, 0x1000)
	_cgo_gotypes.go:90 +0x45 fp=0xc0003ff9f0 sp=0xc0003ff9c0 pc=0x7aa225
github.com/wailsapp/wails/lib/renderer/webview.(*webview).Dialog.func4(0xc000010010, 0x0, 0x1, 0x199c5e0, 0x1b9a060, 0x1c5e000)
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/lib/renderer/webview/webview.go:324 +0x9b fp=0xc0003ffa38 sp=0xc0003ff9f0 pc=0x7ac08b
github.com/wailsapp/wails/lib/renderer/webview.(*webview).Dialog(0xc000010010, 0x0, 0x1, 0x89d82d, 0x10, 0x0, 0x0, 0x0, 0x0)
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/lib/renderer/webview/webview.go:324 +0x129 fp=0xc0003ffa90 sp=0xc0003ffa38 pc=0x7ab419
github.com/wailsapp/wails/lib/renderer.(*WebView).SelectDirectory.func1.1()
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/lib/renderer/webview.go:263 +0x77 fp=0xc0003ffaf8 sp=0xc0003ffa90 pc=0x7b16c7
github.com/wailsapp/wails/lib/renderer/webview._webviewDispatchGoCallback(0x2)
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/lib/renderer/webview/webview.go:355 +0xa4 fp=0xc0003ffb30 sp=0xc0003ffaf8 pc=0x7ab734
github.com/wailsapp/wails/lib/renderer/webview._cgoexpwrap_9664ff72fa97__webviewDispatchGoCallback(0x2)
	_cgo_gotypes.go:309 +0x2b fp=0xc0003ffb48 sp=0xc0003ffb30 pc=0x7aabfb
runtime.call32(0x0, 0x7fffb19e93d0, 0x7fffb19e9460, 0x8)
	/snap/go/current/src/runtime/asm_amd64.s:519 +0x3b fp=0xc0003ffb78 sp=0xc0003ffb48 pc=0x45a14b
runtime.cgocallbackg1(0x0)
	/snap/go/current/src/runtime/cgocall.go:314 +0x177 fp=0xc0003ffbf0 sp=0xc0003ffb78 pc=0x406ac7
runtime.cgocallbackg(0x0)
	/snap/go/current/src/runtime/cgocall.go:191 +0xc5 fp=0xc0003ffc58 sp=0xc0003ffbf0 pc=0x4068b5
runtime.cgocallback_gofunc(0x40677f, 0x7b4b60, 0xc0003ffce8, 0xc0003ffcd8)
	/snap/go/current/src/runtime/asm_amd64.s:773 +0x9b fp=0xc0003ffc78 sp=0xc0003ffc58 pc=0x45b71b
runtime.asmcgocall(0x7b4b60, 0xc0003ffce8)
	/snap/go/current/src/runtime/asm_amd64.s:620 +0x42 fp=0xc0003ffc80 sp=0xc0003ffc78 pc=0x45b5b2
runtime.cgocall(0x7b4b60, 0xc0003ffce8, 0x0)
	/snap/go/current/src/runtime/cgocall.go:131 +0x7f fp=0xc0003ffcb8 sp=0xc0003ffc80 pc=0x40677f
github.com/wailsapp/wails/lib/renderer/webview._Cfunc_CgoWebViewLoop(0x1845cf0, 0x1, 0x0)
	_cgo_gotypes.go:182 +0x4d fp=0xc0003ffce8 sp=0xc0003ffcb8 pc=0x7aa6fd
github.com/wailsapp/wails/lib/renderer/webview.(*webview).Loop.func1(0xc000010010, 0xc000000001, 0x0)
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/lib/renderer/webview/webview.go:280 +0x66 fp=0xc0003ffd28 sp=0xc0003ffce8 pc=0x7ab976
github.com/wailsapp/wails/lib/renderer/webview.(*webview).Loop(0xc000010010, 0x1, 0x1)
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/lib/renderer/webview/webview.go:280 +0x34 fp=0xc0003ffd50 sp=0xc0003ffd28 pc=0x7ab004
github.com/wailsapp/wails/lib/renderer/webview.(*webview).Run(0xc000010010)
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/lib/renderer/webview/webview.go:284 +0x30 fp=0xc0003ffd78 sp=0xc0003ffd50 pc=0x7ab050
github.com/wailsapp/wails/lib/renderer.(*WebView).Run(0xc000070060, 0xa4d160, 0xc000070060)
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/lib/renderer/webview.go:222 +0x1ba fp=0xc0003ffde0 sp=0xc0003ffd78 pc=0x7af68a
github.com/wailsapp/wails.(*App).start(0xc00047e180, 0xc00000e040, 0x0)
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/app.go:116 +0x440 fp=0xc0003ffea0 sp=0xc0003ffde0 pc=0x7b2b70
github.com/wailsapp/wails.(*App).Run(0xc00047e180, 0x7f7320, 0x9cb040)
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/app.go:74 +0xaf fp=0xc0003fff18 sp=0xc0003ffea0 pc=0x7b25ff
main.main()
	/home/lyimmi/go/src/github.com/lyimmi/test/main.go:47 +0x243 fp=0xc0003fff98 sp=0xc0003fff18 pc=0x7b3ee3
runtime.main()
	/snap/go/current/src/runtime/proc.go:200 +0x20c fp=0xc0003fffe0 sp=0xc0003fff98 pc=0x430fbc
runtime.goexit()
	/snap/go/current/src/runtime/asm_amd64.s:1337 +0x1 fp=0xc0003fffe8 sp=0xc0003fffe0 pc=0x45be61

goroutine 19 [syscall]:
os/signal.signal_recv(0x0)
	/snap/go/current/src/runtime/sigqueue.go:139 +0x9c
os/signal.loop()
	/snap/go/current/src/os/signal/signal_unix.go:23 +0x22
created by os/signal.init.0
	/snap/go/current/src/os/signal/signal_unix.go:29 +0x41

goroutine 35 [chan receive]:
github.com/wailsapp/wails/lib/event.(*Manager).Start.func1(0xc0000241c4, 0xc0004c6090)
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/lib/event/manager.go:101 +0x235
created by github.com/wailsapp/wails/lib/event.(*Manager).Start
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/lib/event/manager.go:95 +0x14b

goroutine 36 [chan receive]:
github.com/wailsapp/wails/lib/ipc.(*Manager).Start.func1(0xc00006e040, 0xa472e0, 0xc00047e100)
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/lib/ipc/manager.go:51 +0xa2
created by github.com/wailsapp/wails/lib/ipc.(*Manager).Start
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/lib/ipc/manager.go:47 +0x14d

goroutine 10 [semacquire]:
sync.runtime_Semacquire(0xc000024268)
	/snap/go/current/src/runtime/sema.go:56 +0x39
sync.(*WaitGroup).Wait(0xc000024260)
	/snap/go/current/src/sync/waitgroup.go:130 +0x65
github.com/wailsapp/wails/lib/renderer.(*WebView).SelectDirectory(0xc000070060, 0x8, 0xc00000e3c0)
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/lib/renderer/webview.go:267 +0xab
github.com/wailsapp/wails/runtime.(*Dialog).SelectDirectory(...)
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/runtime/dialog.go:24
main.(*MyRuntime).SelectDirectory(0xc000010008, 0x0, 0x0)
	/home/lyimmi/go/src/github.com/lyimmi/test/main.go:25 +0x3b
reflect.Value.call(0x825400, 0xc000010008, 0x213, 0x895d46, 0x4, 0xd66840, 0x0, 0x0, 0x2, 0xc000030300, ...)
	/snap/go/current/src/reflect/value.go:447 +0x461
reflect.Value.Call(0x825400, 0xc000010008, 0x213, 0xd66840, 0x0, 0x0, 0x1, 0x4d30d8, 0xc000112000)
	/snap/go/current/src/reflect/value.go:308 +0xa4
github.com/wailsapp/wails/lib/binding.(*boundMethod).call(0xc0000c0080, 0xc00040a080, 0x2, 0x1e, 0xc00009e568, 0xc000407bb0, 0x448b97, 0xc000407c57)
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/lib/binding/method.go:143 +0x6aa
github.com/wailsapp/wails/lib/binding.(*Manager).processMethodCall(0xc00047e100, 0xc00009c280, 0xc000407ce0, 0x19, 0xc000407ca0, 0x1)
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/lib/binding/manager.go:216 +0xa7
github.com/wailsapp/wails/lib/binding.(*Manager).ProcessCall(0xc00047e100, 0xc00009c280, 0x0, 0x0, 0x0, 0x0)
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/lib/binding/manager.go:266 +0x378
github.com/wailsapp/wails/lib/ipc.(*Manager).Start.func1.1(0xa472e0, 0xc00047e100, 0xc00009c280, 0xc00006e040, 0xc000010040)
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/lib/ipc/manager.go:64 +0x59
created by github.com/wailsapp/wails/lib/ipc.(*Manager).Start.func1
	/home/lyimmi/go/pkg/mod/github.com/wailsapp/wails@v0.18.3/lib/ipc/manager.go:63 +0xada

```

**Comment by leaanthony** [comment]
Hi there. Thanks for raising this. I've not had an issue on OSX with dialogs, however [this PR](https://github.com/wailsapp/wails/pull/258/) has been opened to fix OSX dialogs. Could you possibly test and confirm if this solves your problem?

**Comment by Lyimmi** [comment]
Yep, that seems to solve the issue.

![Screenshot from 2019-10-13 04-56-42](https://user-images.githubusercontent.com/8627125/66710308-ed515e00-ed75-11e9-8f37-8781cf910c34.png)

**Comment by leaanthony** [comment]
Merged #258
